control-plane/agents/src/bin/core/controller/io_engine/mod.rs

```diff
@@ -135,6 +136,21 @@ where
     }
 }
 
+#[async_trait]
+pub(crate) trait SnapshotApi {
+    /// Create a nexus snapshot via gRPC.
+    async fn create_nexus_snapshot(
```

Comment:
Why not add generic snapshot trait?
trait SnapshotApi {
create()
}

Reply:
I actually intended to do that originally, but ran into issues implementing generics for nexus and replica. Let me give it a shot again.